### DISCUSSION TITLE
no error/warning for unnecessary generic functions

### DISCUSSION THREAD

**Discussion by SleepyRoy** [vlang/v#11801] [Ideas]
**V version:** 
master

**OS:** 
win10

**What did you do?**
```V
fn foo<T>() {

} 

fn bar<T>(a int) {

}

fn main() {

}

```

**What did you expect to see?**
A warning or error for foo and bar - T is not used anywhere inside function
 
**What did you see instead?**
compiled

**Comment by danieldaeschle** [comment]
you mean T should be underlined as unused?

**Comment by StunxFS** [comment]
> you mean T should be underlined as unused?

That is what it means, that V notifies that the generics are not being used (T in this case):
```v
fn foo<T>() {
      ~~~ // warn: T is not used anywhere
} 

fn bar<T>(a int) {
      ~~~ // warn: T is not used anywhere 
}
```

**Comment by SleepyRoy** [comment]
exactly
